src/helpers/checks.jl

```diff
@@ -180,11 +184,14 @@ function check_gradient(
     p=rand(M),
     X=rand(M; vector_at=p);
     gradient=grad_f(M, p),
-    check_vector=true,
+    check_vector=false,
     throw_error=false,
+    atol::Real=0,
```

Comment:
Maybe `atol` for tangent vector checks shouldn't be 0 by default because very often such checks compare with 0, and `atol` equal to 0 turns such checks into exact equality.

Reply:
For best of cases I would just want to pass down kwargs to the check and have proper defaults in Manifolds...
until then – should we really introduce a new default here in this niche of the ecosystem?

Reply:
OK, we can leave that for later.